src/sentry/snuba/metrics/helpers.py

```diff
@@ -606,3 +608,59 @@ def translate_results(self):
         ]
 
         return groups
+
+
+def metric_object_factory(op, metric_name):
+    return RawMetric(op, metric_name)
+
+
+class MetricsFieldBase(ABC):
+    def __init__(self, op, metric_name):
+        self.op = op
+        self.metric_name = metric_name
+
+    def get_entity(self, **kwargs):
```

Comment:
nit: Should we make these methods `@abstractmethod`s? https://docs.python.org/3/library/abc.html#abc.abstractmethod